app/components/ConverterPage/ConverterCoinSelector.tsx

```diff
@@ -0,0 +1,327 @@
+      };
+    });
+  };
+
+  const chartDataFetch = async () => {
+    const { name, currency, days, interval } = getFetchDetails(
+      coinData?.fetchName,
+      converterChartDurationSelector
+    );
+    const proxyUrl = "https://corsproxy.io/?";
+    const targetUrl = `https://api.coingecko.com/api/v3/coins/${name.toLowerCase()}/market_chart?vs_currency=${currency}&days=${days}&interval=${interval}`;
+    const response = await fetch(proxyUrl + targetUrl);
+    const data = await response.json();
+    setCoinData((prev: any) => {
+      return { ...prev, coinChartData: data };
+    });
+  };
+
+  useEffect(() => {
+    initialFetch();
+  }, []);
+
+  useEffect(() => {
+    chartDataFetch();
+  }, [converterChartDurationSelector]);
+
+  const handleAmountInputChange = (e: any) => {
+    let inputVal = e.target.value;
+
+    if (parseFloat(inputVal) < 0) {
+      setCoinData((prev: any) => ({ ...prev, amount: "0" }));
+      return;
+    }
+
+    const newAmount = new Big(inputVal);
+    setCoinData((prev: any) => ({ ...prev, amount: newAmount.toString() }));
+
+    if (coinData.data && otherCoinData.data) {
+      const currentPrice = new Big(coinData.data.market_data.current_price.usd);
+      const otherPrice = new Big(
+        otherCoinData.data.market_data.current_price.usd
+      );
+      const newOtherAmount = newAmount.times(currentPrice).div(otherPrice);
+
+      setOtherCoinData((prev: any) => ({
+        ...prev,
+        amount: newOtherAmount.toString(),
+      }));
+    }
+  };
+
+  return coinData.data !== null ? (
+    <div className="flex flex-col w-full">
+      <div className="flex w-full mt-2 items-center justify-center">
+        <img src={coinData.data?.image?.small} alt="coin image" />
+        <div className="relative w-full">
+          <div className="flex gap-2">
+            <input
+              autoComplete="off"
+              type="text"
+              name="coin-text-input"
+              className="appearance-none bg-transparent w-full p-2"
+              value={coinData.name}
+              onChange={(e) => {
+                setCoinData((prev: any) => {
+                  return { ...prev, name: e.target.value };
+                });
+              }}
+              onFocus={() => {
+                setIsCoinNameFocused(true);
+              }}
+              onBlur={() => {
+                setTimeout(() => {
+                  setIsCoinNameFocused(false);
+                }, 400);
+              }}
+            />
+          </div>
+          {isCoinNameFocused === true ? (
+            coinData.name.length > 0 ? (
+              <CoinSearchSelector
+                coinData={coinData}
+                isCoinNameFocused={isCoinNameFocused}
+                isCoinAmountFocused={isCoinAmountFocused}
+                setCoinData={setCoinData}
+              />
+            ) : (
+              false
+            )
+          ) : (
+            false
+          )}
```

Comment:
what is this?

Reply:
isCoinNameFocused &&  coinData.name.length > 0 &&  " <CoinSearchSelector
                coinData={coinData}
                isCoinNameFocused={isCoinNameFocused}
                isCoinAmountFocused={isCoinAmountFocused}
                setCoinData={setCoinData}
              />"